surefire-its/src/test/java/org/apache/maven/surefire/its/jiras/Surefire1787JUnit5IT.java

```diff
@@ -109,14 +115,18 @@ public void testNgWithJupiterEngine()
             .executeTest()
             .verifyErrorFree( 2 )
             .verifyTextInLog( "Running pkg.JUnit5Test" )
-            .verifyTextInLog( "Running pkg.TestNGTest" );
+            .verifyTextInLog( "Running pkg.TestNGTest" )
+            .verifyTextInLog(
+                "Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider" );
     }
 
     @Test
     public void junit4Runner()
     {
         unpack( "junit5-runner" )
             .executeTest()
-            .verifyErrorFreeLog();
+            .verifyErrorFree( 1 )
+            .verifyTextInLog( "Running pkg.JUnit5Tests" )
+            .verifyTextInLog( "Using auto detected provider org.apache.maven.surefire.junit4.JUnit4Provider" );
```

Comment:
`JUnit4Provider` is the actual status with the bug in ASM.
It should be `JUnitPlatformProvider` after the final fix. WDYT?

Reply:
It is ok IMHO. When we want Junit5 test using Junit4 we add junit-platform-runner as a dependency.
https://junit.org/junit5/docs/current/user-guide/#running-tests-junit-platform-runner